### DISCUSSION TITLE
TimeZoneConverter plugin

### DISCUSSION THREAD

**Discussion by emanuel-v-r** [quartznet/quartznet#936] [General] [answered]
Hello,

I have realized that quartznet has this plugin https://www.quartz-scheduler.net/documentation/quartz-3.x/packages/timezoneconverter-integration.html that allows to work with timezones both in windows and unix systems.
I would like to know when  should I use this plugin instead of using the timezone converter lib directly (https://github.com/mj1856/TimeZoneConverter). I know that the plugin uses this library under the hoods but I am asking this because I am using this in my triggers `WithCronSchedule(cron, opts => opts.InTimeZone(TZConvert.GetTimeZoneInfo("Eastern Standard Time")))` and it seems to work both in windows and linux.
So if this works well, when should I use the plugin?
Probably I am missing something and/or doing unnecessary manual work, so I would really appreciate if someone could point me into the right direction.
Thanks in  advance.

Regards

**Comment by lahma** [comment, marked as answer]
You can certainly use the TimeZoneConverter library directly if you wish to do so. The Quartz plugin is just there to advertise the "better" option which handles cross-OS identifiers better. 

You need to ensure that you hook the resolver properly so Quartz will use it instead of it default. Otherwise you end up having ids that Quartz might not be able to resolve without the help of `TZConvert`.

https://github.com/quartznet/quartznet/blob/effbc3ee162e429ea6f198cc67927cdf01747c55/src/Quartz.Plugins.TimeZoneConverter/Plugin/TimeZoneConverter/TimeZoneConverterPlugin.cs#L46

So basically the plugin offers the hooking and a fast way to configure your setup (if you use the fluent configuration builder):

https://github.com/quartznet/quartznet/blob/effbc3ee162e429ea6f198cc67927cdf01747c55/src/Quartz.Examples.AspNetCore/Startup.cs#L133-L134

**Reply by emanuel-v-r** [reply]
> You can certainly use the TimeZoneConverter library directly if you wish to do so. The Quartz plugin is just there to advertise the "better" option which handles cross-OS identifiers better.
> 
> You need to ensure that you hook the resolver properly so Quartz will use it instead of it default. Otherwise you end up having ids that Quartz might not be able to resolve without the help of `TZConvert`.
> 
> https://github.com/quartznet/quartznet/blob/effbc3ee162e429ea6f198cc67927cdf01747c55/src/Quartz.Plugins.TimeZoneConverter/Plugin/TimeZoneConverter/TimeZoneConverterPlugin.cs#L46
> 
> So basically the plugin offers the hooking and a fast way to configure your setup (if you use the fluent configuration builder):
> 
> https://github.com/quartznet/quartznet/blob/effbc3ee162e429ea6f198cc67927cdf01747c55/src/Quartz.Examples.AspNetCore/Startup.cs#L133-L134

Thanks for your help.
So just to understand, I am not getting any problem just because I am converting already the dates manually? Otherwise QuartzNet would convert it automatically and therefore I would have some error?

**Reply by lahma** [reply]
If you use Windows time zone ids known to .NET there should be no problem, if there's id that doesn't resolve without TZConvert you need to hook the CustomResolver.

**Reply by emanuel-v-r** [reply]
> If you use Windows time zone ids known to .NET there should be no problem, if there's id that doesn't resolve without TZConvert you need to hook the CustomResolver.

That's weird, I needed to use TZConvert for unix (as I showed above), but I didn't plug the plugin and everything seems to be ok.
I will hook the plugin just for safety. Thank you!
